app/models/topology.rb

```diff
@@ -92,4 +93,14 @@ def validate_stakeholders
       end
     end
   end
+
+  def validate_clone
+    return unless original.present?
+    return unless each_node
```

Comment:
What's the intent of this line? `each_node` [will return false if there are already errors present on the topology](https://github.com/quintel/etmoses/blob/b4ee9c19f6cf8622fa75ac543807b6128a4ca722/app/models/topology.rb#L34), but will returns a truthy value in all other situations.

If that's really what you want, I think it best to check explicitly for that instead of relying on the behaviour of `each_node`.

```ruby
def validate_clone
  return if original.blank? || errors[:graph].any?
  # ...
end
```